### PR TITLE
remove use internal method

### PR DESCRIPTION
**Pull request by alavieille**

### CODE REVIEW

File: Pagination/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategy.php [hunk 1]
```diff
@@ -61,7 +61,7 @@ public function generateFilter($name, $value, $documentName)
         if (2 == count($columnsTree)) {
             list($property, $referenceProperty) = $columnsTree;
 
-            $metadata = $this->documentManager->getClassMetadata($documentName);
+            $metadata = $this->documentManager->getMetadataFactory()->getMetadataFor($documentName);
```
Comment (nicolasThal) [inline Pagination/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategy.php]: Why would you suppress the internal call ?

This way you must be aware of the metadatFactory (which should not be the case )

**Comment by alavieille** [comment]
@nicolasThal

**Comment by nicolasThal** [comment]
:+1:

**Comment by ghost** [comment]
:+1: for the test file update

**Pull request merged**